search_service/proxy/gcp_data_catalog.py

```diff
@@ -0,0 +1,248 @@
+    def _process_table_resource(self, entry: str) -> Table:
+        linked_resource_parts = entry.linked_resource.split('/')
+        relative_resource_name_parts = entry.relative_resource_name.split('/')
+
+        name = linked_resource_parts[-1]
+
+        _database = entry.user_specified_system or entry.integrated_system
+
+        if isinstance(_database, int):
+            if _database == 1:
+                database = 'bigquery'
+            schema = linked_resource_parts[-3]
+        else:
+            database = _database
+            schema = relative_resource_name_parts[-1].replace(name, '').strip('_')
+
+        cluster = relative_resource_name_parts[1] + '__' + relative_resource_name_parts[3]
+
+        # @todo handle tags and badges/labels
+        # for tag in self.client.list_tags(entry.relative_resource_name):
+        #     for field in tag.fields:
+        #         value = f"{value['display_name']}:{value['string_value']}"
+        #
+        #         tags.append(value)
+
+        tags = []
+        badges = tags
+
+        result = Table(database=database,
+                       cluster=cluster,
+                       schema=schema,
+                       name=name,
+                       tags=tags,
+                       badges=badges,
+                       last_updated_timestamp=None,
+                       column_names=[],
+                       key=entry.relative_resource_name)
+
+        return result
+
+    def _process_dashboard_resource(self, entry: str) -> Dashboard:
+        full_entry = self.client.get_entry(entry.relative_resource_name)
+
+        dashboard_metadata = self._get_resource_metadata(full_entry.name, r'.*Dashboard Metadata$')
+
+        relative_resource_name_parts = full_entry.name.split('/')
+        result = Dashboard(cluster=relative_resource_name_parts[1],
+                           group_name=dashboard_metadata.get('site_name'),
+                           group_url='',
+                           url='',
+                           product=entry.user_specified_system or entry.integrated_system,
+                           name=dashboard_metadata['workbook_name'],
+                           last_successful_run_timestamp=0,
+                           uri=full_entry.name)
+
+        return result
+
+    def _basic_search(self, query_term: str, page_index: int, entry_type: str = None, **additional_filters):
+        entries = []
+        i = 0
+
+        filters = additional_filters
+        filters['type'] = entry_type
+
+        query = f'{query_term}'
+
+        # different fields support different separators in search dsl syntax
+        # fields like type and system support only exact matching so '=' separator is required
+        # for remaining fields we use pattern match with ':' separator
+        separators = {'type': '=', 'system': '='}
+
+        for field, values in filters.items():
+            separator = separators.get(field, ':')
+
+            if field == 'label':
```

Comment:
I would introduce an inline function which would return (label_name, label_value) to make this part of a code clearer and shorter. Also having a separate function get_query might improve readability and hide complexity